lib/src/main/java/ECSE428Project/controller/AccountController.java

```diff
@@ -99,6 +99,51 @@ public AccountDto updateAccountWithNewEmail(@RequestParam("oldEmail") String old
         return convertToDto(account);
     }
 
+    /**
+     * This method updates the score associated to an account
+     * @param email the email associated to the account
+     * @param scoreString the score that is going to be assigned to the account as a String
+     * @return updated account
+     * @throws ResponseStatusException if the score is not a number
+     */
+    @PutMapping(path = "/account/email/score")
+    public AccountDto assignScoreToAccount(@RequestParam("email") String email, @RequestParam("score") String scoreString)
+            throws ResponseStatusException {
+
+        try {
+            Double.parseDouble(scoreString);
+        } catch (NumberFormatException error) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The score is not a number.");
+        }
+
+        return null;
+        // These lines will be uncommented when the service method is added
+        //Account account = accountService.assignScoreToAccount(email, Double.parseDouble(scoreString));
+        //return convertToDto(account);
+    }
+
+    /**
+     * This method updates the rank associated to an account
+     * @param email the email associated to the account
+     * @param rankString the rank that is going to be assigned to the account as a String
+     * @return updated account
+     * @throws ResponseStatusException if the rank is not a number
+     */
+    @PutMapping(path = "/account/email/rank")
+    public AccountDto assignRankToAccount(@RequestParam("email") String email, @RequestParam("rank") String rankString)
+            throws ResponseStatusException {
+        try {
+            Double.parseDouble(rankString);

```

Comment:
same here

Reply:
For rank I just thought might as well use a double since we can convert that to an int later on but we won't be able to do it the other way around

Reply:
ok but i just dont think it's super necessary to have it as a double at any point

Reply:
Yeah you're right I'll change that